src/xmipp/libraries/py_xmipp/deepDenoising/dataGenerator.py

```diff
@@ -0,0 +1,189 @@
+import pyworkflow.em.metadata as md
+import xmippLib
+import numpy as np
+import random
+
+from sklearn.utils import shuffle
+from sklearn.cross_validation import train_test_split
+from .augmentators import (_random_flip_leftright, _random_flip_updown, _mismatch_projection, 
+                          _random_90degrees_rotation, _random_rotation,generateEmptyParticlesFunction)
+
+BATCH_SIZE= 16
+def getDataGenerator( imgsMdXmd, masksMdXmd, xmdEmptyParts=None, augmentData=True, nEpochs=-1, isTrain=True, valFraction=0.1, 
```

Comment:
very long method

Reply:
It is 100 lines, and if you look at it, inside, some other auxiliary functions are defined, so I do not think it is too long